From c771ce5af61dd19afdb18353ae4453c77d3fe0d6 Mon Sep 17 00:00:00 2001
From: kanlihu <kanli.hu@intel.com>
Date: Sat, 9 Apr 2022 01:33:33 +0800
Subject: [PATCH] i915/drm: enable multi-plane, force cursor spirte to be
 device plane

add prop to /vendor/default.prop
vendor.hwcomposer.planes.enabling=1
to enable mult-plane

Tracked-On: OAM-99951
Signed-off-by: Kanli Hu <kanli.hu@intel.com>
---
 backend/Backend.cpp | 2 +-
 drm/DrmDevice.cpp   | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/backend/Backend.cpp b/backend/Backend.cpp
index 44afd14..cf2d4a7 100644
--- a/backend/Backend.cpp
+++ b/backend/Backend.cpp
@@ -112,7 +112,7 @@ uint32_t Backend::CalcPixOps(const std::vector<DrmHwcTwo::HwcLayer *> &layers,
 void Backend::MarkValidated(std::vector<DrmHwcTwo::HwcLayer *> &layers,
                             size_t client_first_z, size_t client_size) {
   for (int z_order = 0; z_order < layers.size(); ++z_order) {
-    if (z_order >= client_first_z && z_order < client_first_z + client_size)
+    if ((layers[z_order]->sf_type() != HWC2::Composition::Cursor) && z_order >= client_first_z && z_order < client_first_z + client_size)
       layers[z_order]->set_validated_type(HWC2::Composition::Client);
     else
       layers[z_order]->set_validated_type(HWC2::Composition::Device);
diff --git a/drm/DrmDevice.cpp b/drm/DrmDevice.cpp
index d7bc13a..8339c07 100644
--- a/drm/DrmDevice.cpp
+++ b/drm/DrmDevice.cpp
@@ -339,8 +339,10 @@ std::tuple<int, int> DrmDevice::Init(const char *path, int num_displays) {
       if (plane->type() == DRM_PLANE_TYPE_PRIMARY)
         planes_.emplace_back(std::move(plane));
     } else {
-      planes_.emplace_back(std::move(plane));
-    }
+       if (plane->type() != DRM_PLANE_TYPE_CURSOR) {
+		  planes_.emplace_back(std::move(plane));
+        }
+      }
   }
   drmModeFreePlaneResources(plane_res);
   if (ret)
-- 
2.31.0

